File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    total_iron = 0

    for i in range(1, number + 1):
        count = 0
        for j in range(1, int(i ** 0.5) + 1):
            if i % j == 0:
                if j * j == i:
                    count += 1 
                else:
                    count += 2

        if count > limit:
            total_iron += power
        else:
            total_iron += count

    return total_iron

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    let totalIron = 0;
    
    for (let i = 1; i <= number; i++) {
        let count = 0;
        
        for (let j = 1; j * j <= i; j++) {
            if (i % j == 0) count += (j * j === i) ? 1 : 2;
        }
        if (count > limit) totalIron += power;
        else totalIron += count;
    }
    
    return totalIron;
}